http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java

```diff
@@ -372,6 +375,38 @@ public HttpClientConfig setHttpsExcludedCipherSuites(String excludedCipherSuites
         return this;
     }
 
+    public String getAutomaticHttpsSharedSecret()
+    {
+        return automaticHttpsSharedSecret;
+    }
+
+    @ConfigSecuritySensitive
+    @Config("http-client.https.automatic-shared-secret")
+    public HttpClientConfig setAutomaticHttpsSharedSecret(String automaticHttpsSharedSecret)
+    {
+        this.automaticHttpsSharedSecret = automaticHttpsSharedSecret;
+        return this;
+    }
+
+    public String getNodeEnvironment()
+    {
+        return nodeEnvironment;
+    }
+
+    // this is only here to simplify setup
+    @Config("node.environment")
+    public HttpClientConfig setNodeEnvironment(String nodeEnvironment)
+    {
+        this.nodeEnvironment = nodeEnvironment;
+        return this;
+    }
+
+    @AssertTrue(message = "Node environment must be provided when automatic HTTPS is enabled")
+    public boolean isHttpsConfigurationValid()
+    {
+        return getAutomaticHttpsSharedSecret() == null || getNodeEnvironment() != null;
```

Comment:
If we use `NodeInfo`, then this condition should not be possible, since environment is a required configuration